fili-core/src/main/java/com/yahoo/bard/webservice/data/config/table/BaseTableLoader.java

```diff
@@ -57,33 +60,23 @@ protected BaseTableLoader() {
         this(DateTimeZone.UTC);
     }
 
-    @Override
-    public abstract void loadTableDictionary(ResourceDictionaries dictionaries);
-
     /**
-     * Builds a table group.
-     * <p>
-     * Builds and loads the physical tables for the physical table definitions as well.
-     *
-     * @param logicalTableName  The logical table for the table group
-     * @param apiMetrics  The set of metric names surfaced to the api
-     * @param druidMetrics  Names of druid datasource metric columns
-     * @param tableDefinitions  A list of config objects for physical tables
-     * @param dictionaries  The container for all the data dictionaries
+     * Load user configured tables into resource dictionary.
      *
-     * @return A table group binding all the tables for a logical table view together.
+     * @param dictionaries dictionary to be loaded with user configured tables
      *
-     * @deprecated logicalTableName is not used in TableGroup
+     * @deprecated should use the same method that takes an additional DataSourceMetadataService argument instead
      */
     @Deprecated
-    public TableGroup buildTableGroup(
-            String logicalTableName,
-            Set<ApiMetricName> apiMetrics,
-            Set<FieldName> druidMetrics,
-            Set<PhysicalTableDefinition> tableDefinitions,
-            ResourceDictionaries dictionaries
+    public abstract void loadTableDictionary(ResourceDictionaries dictionaries);
```

Comment:
If we're going to be deprecating this, then we shouldn't make it abstract. We really shouldn't be _forcing_ people to implement a deprecated method. 

If we still want to force people to implement _something_, then we should make the new `loadTableDictionary` abstract, and include a small snippet in the migration path that basically tells users to do what we do below.

It's a little bit of pain for people upgrading to the new version, but I would argue it's better than forcing people to implement cruft, and it's a very minor compile-time pain so whatever.